osquery/tables/system/darwin/shared_folders.mm

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <OpenDirectory/OpenDirectory.h>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+Status genSharePointEntries(std::vector<NSDictionary *> &sharepoints) {
```

Comment:
It's worth noting that, after everything that happens in `include/osquery/database.h`, `osquery::QueryData` is also just a `std::vector<std::map<string, string>>`, so with some different types, you could just accept a `osquery::QueryData` instead of a `std::vector`, and then you could just return that. At that point, your other function is basically your gen function.